src/coreclr/src/tools/Common/Compiler/CompilerTypeSystemContext.cs

```diff
@@ -118,7 +118,7 @@ public ModuleDesc GetModuleForSimpleName(string simpleName, bool throwIfNotFound
                 if (!ReferenceFilePaths.TryGetValue(simpleName, out filePath))
                 {
                     // We allow the CanonTypesModule to not be an EcmaModule.
-                    if (((IAssemblyDesc)CanonTypesModule).GetName().Name == simpleName)
+                    if (CanonTypesModule != null && ((IAssemblyDesc)CanonTypesModule).GetName().Name == simpleName)
```

Comment:
Should we rather throw earlier when there is no system module?

Something like https://github.com/dotnet/corert/pull/7972 ?

cc @MichalStrehovsky

Reply:
Great, that PR addresses the issue, so I will revert this part.

Reply:
Oh, that is a CoreRT fix. I may add the same code for crossgen2.

Reply:
Reverted this change in favor of a better fix — need to port dotnet/corert#7972.

Reply:
Yep, checking earlier is better than trying to make the system resilient to bad state.

I never liked the `SetSystemModule` API. It's a wart. Type system context shouldn't have mutable state that makes it possible to use it before it's ready to be used.